osu.Game/Rulesets/Objects/SliderPathExtensions.cs

```diff
@@ -14,26 +17,28 @@ public static class SliderPathExtensions
         /// <summary>
         /// Snaps the provided <paramref name="hitObject"/>'s duration using the <paramref name="snapProvider"/>.
         /// </summary>
-        public static void SnapTo<THitObject>(this THitObject hitObject, IDistanceSnapProvider? snapProvider)
+        public static void SnapTo<THitObject>(this THitObject hitObject, IDistanceSnapProvider? snapProvider, NewBeatmapEditorChangeHandler? changeHandler = null)
             where THitObject : HitObject, IHasPath
         {
-            hitObject.Path.ExpectedDistance.Value = snapProvider?.FindSnappedDistance(hitObject, (float)hitObject.Path.CalculatedDistance, DistanceSnapTarget.Start) ?? hitObject.Path.CalculatedDistance;
+            double newDistance = snapProvider?.FindSnappedDistance(hitObject, (float)hitObject.Path.CalculatedDistance, DistanceSnapTarget.Start) ?? hitObject.Path.CalculatedDistance;
+            new ExpectedDistanceChange(hitObject.Path, newDistance).Submit(changeHandler);
         }
 
         /// <summary>
         /// Reverse the direction of this path.
         /// </summary>
         /// <param name="sliderPath">The <see cref="SliderPath"/>.</param>
         /// <param name="positionalOffset">The positional offset of the resulting path. It should be added to the start position of this path.</param>
-        public static void Reverse(this SliderPath sliderPath, out Vector2 positionalOffset)
+        /// <param name="changeHandler">Change handler to submit changes to.</param>
+        public static void Reverse(this SliderPath sliderPath, out Vector2 positionalOffset, NewBeatmapEditorChangeHandler? changeHandler = null)
```

Comment:
In line with what I said previously, the feeling I have here is that this helper method should really eventually just become a command.

Same thing goes for every user operation that ends up submitting 2+ commands.

Reply:
I'll turn these into some kind of commands.